src/components/StyledListItem.js

```diff
@@ -0,0 +1,48 @@
+import {
+  CategoryAndDate,
+  LeftBlock,
+  RightBlock,
+  StyledAmount,
+  StyledCategory,
+  StyledDate,
+  StyledLink,
+} from "@/src/components/styles/List";
+import { CategoryIcon } from "@/src/components/CategoryIcon";
+
+// Utility function to get currency symbol
+const getCurrencySymbol = (currencyCode) => {
+  const symbols = {
+    USD: "$",
+    EUR: "€",
+    GBP: "£",
+    JPY: "¥",
+  };
+  return symbols[currencyCode] || currencyCode;
+};
+
+export function StyledListItem({ transaction }) {
+
+  return (
+    <StyledLink href={`/transactions/${transaction.id}`}>
+      <LeftBlock>
+        <CategoryIcon category={transaction.category} />
+        <CategoryAndDate>
+          <StyledCategory>{transaction.category}</StyledCategory>
+          <StyledDate>
+            {new Date(transaction.date).toLocaleDateString("en-GB", {
+              day: "2-digit",
+              month: "2-digit",
+              year: "numeric",
+            }).split("/").join(".")}
+          </StyledDate>
+        </CategoryAndDate>
+      </LeftBlock>
+      <RightBlock>
+        <StyledAmount>
+          {transaction.amount.toFixed(2)}{" "}
+          {getCurrencySymbol(transaction.currency)}
+        </StyledAmount>
+      </RightBlock>
+    </StyledLink>
+  );
+}
```

Comment:
Maybe more a QA thing and you probably handle this somewhere else, but if I add a new transaction, I always get an additional row with "Invalid Date - NaN"

<img width="513" alt="Bildschirmfoto 2024-06-13 um 16 08 53" src="https://github.com/miToe/dagobert/assets/163883411/f7b5764e-26c6-476e-a0c5-24fb0199bbfa">

Reply:
Yes you're right, this will be taken care of in another ticket, which is in QA right now.